### DISCUSSION TITLE
date style versioning

### DISCUSSION THREAD

**Discussion by yfhyou** [renovatebot/renovate#26604] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Is it possible to have a date oriented version?
I am trying to update a dockerfile that uses debian:bookworm-20231218-slim which isn't covered by the default rules. I tried setting up a custom package rule:
```
  "packageRules": [
    {
      "matchPackageNames": ["debian"],
      "versioning": "regex:^(?<major>.*)\\-(?<build>\\d+)-(?<compatibility>.*)$"
    }
  ]
```
My thought is that 'build' would just get updated with the next highest number, in this case `bookworm-20240110-slim`, but it doesn't seem to just increment to the next highest. I know this isn't specifically supported, but I thought it could be worked out with a regex and custom package rule.

Did I do my regex incorrectly, or this isn't possible in this manner?

### Logs (if relevant)

<details><summary>Logs</summary>

```
...
DEBUG: packageFiles with updates
{
  "baseBranch": "dev"
  "config": {
    "dockerfile": [
      {
        "deps": [
          {
            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "currentDigest": "sha256:f80c45482c8d147da87613cb6878a7238b8642bcc24fc11bad78c7bec726f340",
            "currentValue": "bookworm-20231218-slim",
            "currentVersion": "bookworm-20231218-slim",
            "datasource": "docker",
            "depName": "debian",
            "depType": "final",
            "fixedVersion": "bookworm-20231218-slim",
            "packageName": "debian",
            "registryUrl": "https://index.docker.io",
            "replaceString": "debian:bookworm-20231218-slim@sha256:f80c45482c8d147da87613cb6878a7238b8642bcc24fc11bad78c7bec726f340",
            "versioning": "regex:^(?<major>.*)\\-(?<build>\\d+)-(?<compatibility>.*)$",
            "warnings": [],
            "updates": []
          }
        ],
        "packageFile": "Dockerfile"
      }
    ],
    "github-actions": [
...
```

</details>

**Comment by secustor** [comment, marked as answer]
Use the year as major, month as minor and days as patch version.

**Reply by yfhyou** [reply]
This is great! I never thought to break the date down in that way. Thank you :)

I ended up with this for anyone looking for something similiar:
```
  "packageRules": [
    {
      "matchPackageNames": ["debian"],
      "versioning": "regex:^bookworm-(?<major>\\d{4})(?<minor>\\d{2})(?<patch>\\d{2})-(?<compatibility>.*)$"
    }
  ]
```

**Comment by yfhyou** [comment]
> Use the year as major, month as minor and days as patch version. 

Out of curiosity, is there a digit limit to the versions? Is that why doing it as one didn't work?
